lang/java/avro/src/test/java/org/apache/avro/TestSchemaNormalization.java

```diff
@@ -60,6 +71,88 @@ public void testCanonicalization() throws Exception {
     }
   }
 
+  @RunWith(Parameterized.class)
+  public static class TestStandardCanonicalSchema {
+    String input, expectedOutput;
+
+    public TestStandardCanonicalSchema(String i, String o) {
+      input = i;
+      expectedOutput = o;
+    }
+
+    @Parameters
+    public static List<Object[]> cases() throws IOException {
+      return CaseFinder.find(data(STANDARD_CANONICAL_DATA_FILE), "canonical", new ArrayList<>());
+    }
+
+    @Test
+    public void testCanonicalization() throws Exception {
+      assertEquals(SchemaNormalization.toCanonicalForm(new Schema.Parser().parse(input)), expectedOutput);
+    }
+  }
+
+  @RunWith(Parameterized.class)
+  public static class TestCustomCanonicalSchema {
+    String input, expectedOutput;
+    LinkedHashSet<String> properties = new LinkedHashSet<>(Arrays.asList("format"));
+
+    public TestCustomCanonicalSchema(String i, String o) {
+      input = i;
+      expectedOutput = o;
+    }
+
+    @Parameters
+    public static List<Object[]> cases() throws IOException {
+      return CaseFinder.find(data(CUSTOM_CANONICAL_DATA_FILE), "canonical", new ArrayList<>());
+    }
+
+    @Test
+    public void testCanonicalization() throws Exception {
+      assertEquals(SchemaNormalization.toCanonicalForm(new Schema.Parser().parse(input), properties), expectedOutput);
+    }
+  }
+
+  @RunWith(Parameterized.class)
```

Comment:
This class is a duplicate of the above and causes a compilation error.

Reply:
duplicate class removed. Thanks @cutting .